src/components/NavBar.tsx

```diff
@@ -105,6 +112,28 @@ export function NavBar({ links }: NavBarProps) {
     return null;
   }
 
+  const isActive = (href: string) =>
+    location.pathname === href ||
+    (isHabitTrackerActive &&
+      (href.includes(RoutePaths.HABITS) ||
+        href.includes(RoutePaths.CALENDAR) ||
+        href.includes(RoutePaths.ANALYTICS)));
+
+  const renderIcon = (text: string) => {
```

Comment:
instead of this, you can add an `icon` key value pair to the `links` object defined in App.tsx